### PR TITLE
Upgrade wrapper to Gradle 7.0 for user projects, support JDK 16 and remove unsupported Java versions

### PR DESCRIPTION
**Pull request by graemerocher**
This PR 

- Upgrades the wrapper to Gradle 7.0
- Supports JDK 16
- Retains only JDKs 8, 11 and 16 since we shouldn't be suggesting unsupported JDKs to users.

### CODE REVIEW

**Comment by graemerocher** [comment]
@ilopmar this is ready for review, the failures seem to be unrelated to this change with one of them being a port binding and another being a problem downloading the `micronaut-core-reactive` POM

File: starter-core/src/main/java/io/micronaut/starter/feature/Features.java [hunk 1]
```diff
@@ -99,11 +99,7 @@ public JdkVersion javaVersion() {
     }
 
     public String getTargetJdk() {
-        if (language().isKotlin() || testFramework().isKoTest()) {
-            return VersionInfo.toJdkVersion(Math.min(javaVersion.majorVersion(), 14));
-        } else {
-            return VersionInfo.toJdkVersion(javaVersion.majorVersion());
-        }
+        return VersionInfo.toJdkVersion(javaVersion.majorVersion());
```
Comment (ilopmar) [inline starter-core/src/main/java/io/micronaut/starter/feature/Features.java]: I've tried a generated Kotlin app with Java 16 and it doesn't work. It fails with

```
> Task :kaptGenerateStubsKotlin
e: Unknown JVM target version: 16
Supported versions: 1.6, 1.8, 9, 10, 11, 12, 13, 14, 15

> Task :kaptGenerateStubsKotlin FAILED

FAILURE: Build failed with an exception.
```

File: starter-core/src/main/java/io/micronaut/starter/feature/graalvm/GraalVMFeatureValidator.java [hunk 2]
```diff
@@ -41,8 +41,8 @@ public void validatePostProcessing(Options options, ApplicationType applicationT
                 throw new IllegalArgumentException("GraalVM is not supported in Groovy applications");
             }
 
-            if (options.getJavaVersion().majorVersion() > JdkVersion.JDK_11.majorVersion()) {
-                throw new IllegalArgumentException("GraalVM only supports up to JDK 11");
+            if (options.getJavaVersion().majorVersion() > JdkVersion.JDK_16.majorVersion()) {
+                throw new IllegalArgumentException("GraalVM only supports up to JDK 16");
```
Comment (ilopmar) [inline starter-core/src/main/java/io/micronaut/starter/feature/graalvm/GraalVMFeatureValidator.java]: We still doesn't support JDK 16 for GraalVM: https://github.com/oracle/graal/issues/3310

File: starter-core/src/main/java/io/micronaut/starter/options/JdkVersion.java [hunk 3]
```diff
@@ -27,13 +27,8 @@
...
 public enum JdkVersion {
     JDK_8(8),
-    JDK_9(9),
-    JDK_10(10),
     JDK_11(11),
-    JDK_12(12),
-    JDK_13(13),
-    JDK_14(14),
-    JDK_15(15);
+    JDK_16(16);
```
Comment (ilopmar) [inline starter-core/src/main/java/io/micronaut/starter/options/JdkVersion.java:31]: :100: :tada:

File: starter-core/src/test/groovy/io/micronaut/starter/feature/graalvm/GraalVMSpec.groovy [hunk 4]
```diff
@@ -118,23 +118,6 @@ class GraalVMSpec extends ApplicationContextSpec implements CommandOutputFixture
         e.message == 'GraalVM is not supported in Groovy applications'
     }
 
-    @Unroll
-    void 'it is not possible to use graalvm with JDK versions different than JDK8 through JDK11'(JdkVersion jdkVersion) {
-        when:
```
Comment (ilopmar) [inline starter-core/src/test/groovy/io/micronaut/starter/feature/graalvm/GraalVMSpec.groovy:123]: If you rollback the change I mentioned before about not supporting Java 16 with GraalVM yet, then you need to bring this test back.

**Review by ilopmar — CHANGES_REQUESTED** [review]

File: starter-core/src/test/groovy/io/micronaut/starter/feature/graalvm/GraalVMSpec.groovy (hunk 4 above)
Comment (ilopmar) [reply starter-core/src/test/groovy/io/micronaut/starter/feature/graalvm/GraalVMSpec.groovy:123]: I still see this tests removed

**Review by ilopmar — APPROVED** [review]

**Pull request merged**